addons/crm/tests/test_crm_lead_duplicated.py

```diff
@@ -0,0 +1,182 @@
+from odoo.tests.common import tagged
+from odoo.addons.crm.tests.common import TestCrmCommon
+
+
+@tagged('lead_duplicates')
+class TestLeadConvert(TestCrmCommon):
+
+    def check_duplicates(self, leads, k, keys):
+        """
+        :param leads: Dictionnary of leads.
+        :param k: Key of the considered lead
+        :param keys: Identifiers of the phone numbers that should appear in the search result.
+        """
+
+        leads[k]._compute_potential_lead_duplicates()
+        fun = lambda elem: elem.id
+        self.assertEqual(
+            sorted([leads[key] for key in keys], key=fun),
+            sorted(list(filter(
+                lambda lead: lead in leads.values(),
+                leads[k].duplicate_lead_ids
+            )), key=fun)
+        )
+
+    def test_potential_lead_duplicates(self):
```

Comment:
Could be interesting to create data in batch to reduce size of diff while being explicit in differences between them. Purpose is to both reduce diff and improve understanding of your test data.